R/db-mssql.R

```diff
@@ -0,0 +1,152 @@
+#' @export
+`db_desc.Microsoft SQL Server` <- function(x) {
+  info <- DBI::dbGetInfo(x)
+  host <- if (info$host == "") "localhost" else info$host
+
+  paste0("Microsoft SQL Server ", info$serverVersion, " [", info$user, "@",
+         host, ":", info$port, "/", info$dbname, "]")
+}
+
+#' @export
+`sql_select.Microsoft SQL Server`<- function(con, select, from, where = NULL,
+                                             group_by = NULL, having = NULL,
+                                             order_by = NULL,
+                                             limit = NULL,
+                                             distinct = FALSE,
+                                             ...) {
+  out <- vector("list", 7)
+  names(out) <- c("select", "from", "where", "group_by",
+                  "having", "order_by","limit")
+
+  assert_that(is.character(select), length(select) > 0L)
+  out$select <- build_sql(
+    "SELECT ",
+
+    if (!is.null(limit) && !identical(limit, Inf)) {
```

Comment:
It would be useful to have a comment here describing where SQL server deviates from SQL-92